lib/appsignal/hooks/puma.rb

```diff
@@ -33,5 +37,59 @@ def stop_workers
         end
       end
     end
+
+    class PumaProbe
+      attr_reader :hostname
+
+      def initialize
+        @hostname = Appsignal.config[:hostname] || Socket.gethostname
+      end
+
+      def call
+        return unless ::Puma.stats
+
+        stats = JSON.parse Puma.stats, :symbolize_names => true
+        counts = {
+          :backlog => 0,
+          :running => 0,
+          :pool_capacity => 0,
+          :max_threads => 0
+        }
+
+        # Multiple workers
+        if stats[:worker_status]
+          stats[:worker_status].each do |worker|
+            stat = worker[:last_status]
+
+            counts[:backlog] += stat[:backlog]
+            counts[:running] += stat[:running]
+            counts[:pool_capacity] += stat[:pool_capacity]
+            counts[:max_threads] += stat[:max_threads]
+          end
+
+          puma_gauge(stats[:workers], :workers, :kind => :count)
+          puma_gauge(stats[:booted_workers], :workers, :kind => :booted)
+          puma_gauge(stats[:old_workers], :workers, :kind => :old)
+
+        # Single worker
+        else
+          counts[:backlog] += stats[:backlog]
+          counts[:running] += stats[:running]
+          counts[:pool_capacity] += stats[:pool_capacity]
+          counts[:max_threads] += stats[:max_threads]
+        end
+
+        puma_gauge(counts[:backlog], :backlog)
+        puma_gauge(counts[:running], :running)
+        puma_gauge(counts[:pool_capacity], :pool_capacity)
+        puma_gauge(counts[:max_threads], :max_threads)
+      rescue => e
+        Appsignal.logger.warn e
+      end
+
+      def puma_gauge(count, field, tags = {})
```

Comment:
this should a private method. We probably also don't need to test it separately. Didn't do that for the Sidekiq probe